docs/en/observability/monitor-k8s/monitor-k8s-application-performance.asciidoc

```diff
@@ -2,52 +2,220 @@
+like when a single APM Server instance should be reused for any pod that may be allocated to a node.
+
+* Deploy APM Server as a sidecar -- For environments that should not share an APM Server,
+like when directing traces from multiple applications to separate {es} clusters.
 
-Question: Are we sure we want to cover ECK here? Can we point to the ECK docs
-instead? If we try to document all the ways in all the sections, I think users
-might get confused.
+* {apm-server-ref-v}/installing.html[Download and install APM Server] -- The classic, non-Kubernetes option.
 
 [discrete]
-==== On cloud
+=== Step 2: Set up APM Agents
 
-TODO: Describe how to set up APM server on cloud.
+In most cases, setting up APM agents and thereby instrumenting your applications
+is as simple as installing a library and adding a few lines of code.
+See the relevant agent setup instructions for more information:
+
+* {apm-go-ref-v}/getting-started.html[Go agent]
+* {apm-java-ref-v}/setup.html[Java agent]
+* {apm-dotnet-ref-v}/setup.html[.NET agent]
+* {apm-node-ref-v}/set-up.html[Node.js agent]
+* {apm-py-ref-v}/set-up.html[Python agent]
+* {apm-ruby-ref-v}/set-up.html[Ruby agent]
+
+[discrete]
+=== Step 3: (Optional) Attach the agent
+
+Some agents offer the ability to instrument supported technologies without
+any changes to an application image or code.
+To do this, you'll need an https://kubernetes.io/docs/concepts/workloads/pods/init-containers/[init container]
+based on an official Elastic APM docker image.
+
+Before your application starts, copy the agent from the init container into a shared volume.
+For example, with the Java agent:
+
+[source,yml]
+----
+    # ...
+    spec:
+      volumes:
+      - name: elastic-apm-agent <1>
+        emptyDir: {}
+      initContainers:
+      - name: elastic-java-agent
+        image: docker.elastic.co/observability/apm-agent-java:1.12.0 <2>
+        volumeMounts:
+        - mountPath: /elastic/apm/agent
+          name: elastic-apm-agent
+        command: ['cp', '-v', '/usr/agent/elastic-apm-agent.jar', '/elastic/apm/agent'] <3>
+----
+<1> The shared volume
+<2> Build the initContainer from the official Elastic Java agent image
+<3> Copy the agent to the shared volume
+
+The Java command line needs a way to pick up this `javaagent` configuration.
+You can use the standard JVMTI https://docs.oracle.com/javase/8/docs/platform/jvmti/jvmti.html#tooloptions[JAVA_TOOL_OPTIONS] environment variable to do this.
+It doesn't have to be explicitly specified and is picked up automatically by the JVM when it starts.
+
+TIP: For JVMs that don't support this option, you can use any other environment variable --
+either one already defined in your startup script, like `JAVA_OPTS` in some servlet container scripts,
+or add a dedicated empty one that will have no effect if it's not set.
+
+[source,yml]
+----
+      # ...
+      containers:
+      - name: your-app-container
+        env:
```

Comment:
To get this to work, I had to add another volume mount before the `env:` settings. 

```
        volumeMounts: 
        - mountPath: /elastic/apm/agent 
          name: elastic-apm-agent 
  ```

Otherwise I see `CrashLoopBackOff` when I view the pod status.

Reply:
🤦 You're right